storage/atsd/atsd.go

```diff
@@ -130,43 +134,43 @@ func (self deduplicationParamsList) Set(value string) error {
 }
 
 // need to parse duration in toml configuration file
-type duration struct {
-	time.Duration
-}
+// and implement UnmarshalText(text []byte) error method
+type duration time.Duration
 
 func (d *duration) UnmarshalText(text []byte) error {
-	var err error
-	d.Duration, err = time.ParseDuration(string(text))
+	dur, err := time.ParseDuration(string(text))
+	*d = duration(dur)
 	return err
 }
 
-func newStorage(username, password, hostname string, argDbBufferDuration time.Duration) storage.StorageDriver {
+type cadvisorParams struct {
+	IncludeAllMajorNumbers bool     `toml:"store_major_numbers"`
+	UserCgroupsEnabled     bool     `toml:"store_user_cgroups"`
+	PropertyInterval       duration `toml:"property_interval"`
+	SamplingInterval       duration `toml:"sampling_interval"`
+	DockerHost             string   `toml:"docker_host"`
+}
 
+func newStorage(username, password, hostname string, bufferDuration time.Duration) (storage.StorageDriver, error) {
 	innerStorageConfig := atsdStorageDriver.GetDefaultConfig()
 
 	selfConfig := struct {
-		Cadvisor struct {
-			IncludeAllMajorNumbers bool     `toml:"store_major_numbers"`
-			UserCgroupsEnabled     bool     `toml:"store_user_cgroups"`
-			PropertyInterval       duration `toml:"property_interval"`
-			SamplingInterval       duration `toml:"sampling_interval"`
-			DockerHost             string   `toml:"docker_host"`
-		}
+		Cadvisor cadvisorParams
```

Comment:
Yes, but I can't use reference here. Default toml marshaller use plain struct and ignores references.